lib/Conversion/AIEVecToLLVM/AIEVecToLLVM.cpp

```diff
@@ -1092,6 +1239,7 @@ class SRSOpConversion : public mlir::ConvertOpToLLVMPattern<aievec::SRSOp> {
 
     // Integer types
     Value srsIntrOp = nullptr;
+    bool isAIE2p = (aieTarget == "aie2p");
```

Comment:
Same here. As a general rule, we want a different set of patterns for each legalization. It's okay if the patterns are parametrized, and you instantiate with different parameters for different architectures. But those large "if" statements with slight variations here and there, depending on the architecture, spell doom.